bikram.md

```diff
@@ -0,0 +1,95 @@
+1. API Authentication:
+Every registered RESTful API should be assigned a unique Client ID and a Client Secret as a part of an HTTP header. The Client Secret should not be shared. Do not solely rely on a 'Client ID' for authentication.
+
+2	Access Control:
+Not every user has a right to every web service. This is vital, as you don't want administrative web services to be misused. The API key should be sent along as a cookie or body parameter to ensure that privileged collections or actions are properly protected from unauthorized use. Every API needs to be authenticated before it can be used.
+
+3	Masking HTTP Headers:
+Server versioning information or any other sensitive information from the HTTP headers should be removed/masked according to industry best practices. This prevents any form of targeted attacks since the vulnerabilities are mostly specific to the vendors.
+
+4	Session management:
+RESTful web services should use session-based authentication, either by establishing a session token via a POST or by using an API key (client_id + client_secret) as a POST body argument or as a cookie. Usernames, passwords, session tokens, API keys, and sensitive information should not appear in the URL, as this can be captured in web server logs, which makes them intrinsically valuable.
+
+5	Protect HTTP methods:
+RESTful API often use GET (read), POST (create), PUT (replace/update) and DELETE (to delete a record). Not all of these are valid choices for every single resource collection, user, or action. Make sure the incoming HTTP method is valid for the session token/API key and associated resource collection, action, and record.
```

Comment:
In other words: Always allow only the appropriate method and make sure the users are authorized accordingly.